skyportal/handlers/api/annotation.py

```diff
@@ -176,15 +266,28 @@ def put(self, annotation_id):
               application/json:
                 schema: Error
         """
-        a = Annotation.get_if_accessible_by(
-            annotation_id, self.current_user, mode="update", raise_if_none=True
-        )
 
+        try:
+            annotation_id = int(annotation_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid annotation ID. ")
+
+        if associated_resource_type.lower() == "sources":
+            schema = Annotation.__schema__()
+            a = Annotation.get_if_accessible_by(
+                annotation_id, self.current_user, mode="update", raise_if_none=True
```

Comment:
I presume this exception (`raise_if_none`) translates into an AccessError; could you verify?

Reply:
In Baselayer this calls for a `raise AccessError(f"Invalid {cls.__name__} id: {pk}")` so yeah, it looks like it.

Reply:
So you need `except AccessError as e:` to handle it.